### PR TITLE
Fix new clippy lints (Rust `1.67`)

### PR DESCRIPTION
**Pull request by sholderbach**
# Description

Primarily more aggressive use of format strings

Missing `clippy::result_large_err`

### CODE REVIEW

File: crates/nu-command/src/viewers/table.rs [hunk 1]
```diff
@@ -1799,19 +1799,7 @@ enum TableView {
...
-            if !config.use_ansi_coloring || !atty::is(atty::Stream::Stdout) {
-                // Draw the table without ansi colors
-                Some(nu_utils::strip_ansi_string_likely(output))
-            } else {
-                // Draw the table with ansi colors
-                Some(output)
-            }
-        }
-        None => None,
-    }
```
Comment (sholderbach) [inline crates/nu-command/src/viewers/table.rs:1814]: What are you smoking clippy?!

Comment (fdncred) [reply crates/nu-command/src/viewers/table.rs:1814]: That's crazy talk. I added this to my rust bump pr
```
#[allow(clippy::manual_filter)]
```

**Comment by fdncred** [comment]
I'm not sure that this `alias_recursion` test from #4487 is even valid any longer. I don't see the code in parser.rs anymore.

**Comment by sholderbach** [comment]
> I'm not sure that this `alias_recursion` test from #4487 is even valid any longer. I don't see the code in parser.rs anymore.

This is caused by the trippy suggested autofix from clippy, if I tracked that down correctly (basically suddently you get `None` returned and panic).

**Comment by sholderbach** [comment]
Superseded by #7906 that only differs in not including the clippy weirdness.

**Pull request closed without merging**